packages/eas-cli/src/project/publish.ts

```diff
@@ -193,43 +199,62 @@ export function loadMetadata(distRoot: string): Metadata {
   if (metadata.bundler !== 'metro') {
     throw new Error('Only bundles created with Metro are currently supported');
   }
+  const platforms = Object.keys(metadata.fileMetadata);
+  if (platforms.length === 0) {
+    Log.warn('No platforms were exported');
+  }
+  Log.debug(`Loaded ${platforms.length} platform(s): ${platforms.join(', ')}`);
   return metadata;
 }
 
-export async function collectAssetsAsync({
-  inputDir,
-  platforms,
-}: {
-  inputDir: string;
-  platforms: PublishPlatform[];
-}): Promise<CollectedAssets> {
-  const distRoot = await resolveInputDirectoryAsync(inputDir);
-  const metadata = loadMetadata(distRoot);
-
-  const assetsFinal: CollectedAssets = {};
-  for (const platform of platforms) {
-    assetsFinal[platform] = {
+export function filterPlatforms<T extends Partial<Record<Platform, any>>>(
+  record: T,
+  platformFlag: ExpoCLIExportPlatformFlag
+): T {
+  if (platformFlag === 'all') {
+    return record;
+  }
+
+  const platform = platformFlag as Platform;
+
+  if (!record[platform]) {
+    throw new Error(
+      `--platform="${platform}" not found in metadata.json. Available platforms: ${Object.keys(
+        record
+      ).join(', ')}`
+    );
+  }
+
+  return { [platform]: record[platform] } as T;
+}
+
+/** Given a directory, load the metadata.json and collect the assets for each platform. */
+export async function collectAssetsAsync(inputDir: string): Promise<CollectedAssets> {
```

Comment:
Rename the parameter since "inputDir" already means something specific in these files, and the argument shouldn't be that inputDir anymore (in practice, it's distRoot but this function looks like it's intended to take any directory passed to it):
```suggestion
export async function collectAssetsAsync(dir: string): Promise<CollectedAssets> {
```